### DISCUSSION TITLE
How to do invalidate the cache of a .env per project?

### DISCUSSION THREAD

**Discussion by arvindell** [vercel/turborepo#1093] [Help] [answered]
Hello 👋  How can I tell Turbo to check the .env file only for my web packages (which inline environment variables)? 

I'd like to get remote caching working for my teammates and in CI too, however, the only way I do it right now is using .env as globalDependencies, which prevents a lot of cache hits

For example: Say I have two packages: api and website. Each project has a separate environment which is managed with a .env inside the package dir. When I update api/.env, I want to preserve cache artifacts for the website package and vice-versa. 

Is this possible? 🤔
I have tried adding environment variables to the pipeline using $ but this never checks the contents of .env

**Comment by anthonyshew** [comment, marked as answer]
You can use a combination of `dotEnv` and `globalDotEnv` according to your use case.

`dotEnv`: https://turbo.build/repo/docs/reference/configuration#globaldotenv
`globalDotEnv`: https://turbo.build/repo/docs/reference/configuration#dotenv
